Sources/SwiftSyntax/Trivia.swift

```diff
@@ -42,6 +42,28 @@ public struct Trivia: Sendable {
     pieces.isEmpty
   }
 
+  /// The string contents of all the comment pieces with any comments tokens trimmed.
+  public var commentValue: String {
+    var comments = [String]()
+
+    for piece in pieces {
+      switch piece {
+      case .blockComment(let text), .docBlockComment(let text):
+        let text = text.trimmingCharacters(in: "/* ").trimmingAnyCharacters(in: ["\n"])
+        comments.append(String(text))
+
+      case .lineComment(let text), .docLineComment(let text):
+        let text = text.trimmingPrefix("/ ")
```

Comment:
I think we want to check if the `/` are followed by a space in all `.lineComment` pieces in this `Trivia` and if so, trim a single space, if not, not trim any space. For example both

```
// - Task
//   - Subtask
// - Task 2
```

and 

```
//- Task
//  - Subtask
//- Task 2
```

should return

```
- Task
  - Subtask
- Task 2
```

as the comment value.